src/langchain_google_spanner/document_loader.py

```diff
@@ -0,0 +1,348 @@
+
+        Args:
+            instance: The Spanner instance to load data to.
+            database: The Spanner database to load data to.
+            table_name: The table name to load data to.
+            client: The connection object to use. This can be used to customized project id and credentials.
+            content_column: Optional. The name of the content column. Defaulted to the first column.
+            metadata_columns: Optional. This is for user to opt-in a selection of columns to use. Defaulted to use
+                              all columns.
+            store_metadata: If true, extra metadata will be stored in the "langchain_metadata" column.
+            metadata_json_column: Optional. The name of the special JSON column. Defaulted to use "langchain_metadata".
+        """
+        self.instance = instance
+        self.database = database
+        self.table_name = table_name
+        self.content_column = content_column
+        self.metadata_columns = metadata_columns
+        self.metadata_json_column = metadata_json_column or METADATA_COL_NAME
+        self.client = client
+        spanner_instance = self.client.instance(instance)
+        if not spanner_instance.exists():
+            raise Exception("Instance doesn't exist.")
+        spanner_database = spanner_instance.database(database)
+        if not spanner_database.exists():
+            raise Exception("Database doesn't exist.")
+        spanner_database.reload()
+        self.dialect = spanner_database.database_dialect
+        spanner_table = spanner_database.table(table_name)
+        if not spanner_table.exists():
+            raise Exception(
+                "Table doesn't exist. Create table with SpannerDocumentSaver.init_document_table function."
+            )
+        self._table_fields = [
+            n.name for n in spanner_table.schema if n.name != self.metadata_json_column
+        ]
+        self._table_fields.append(self.metadata_json_column)
+
+    def add_documents(self, documents: List[Document]):
+        """Add documents to the Spanner table."""
+        db = self.client.instance(self.instance).database(self.database)
+        values = [
+            _load_doc_to_row(self._table_fields, doc, self.metadata_json_column)
+            for doc in documents
+        ]
+
+        for values_batch in _batch(values, MUTATION_BATCH_SIZE):
+            with db.batch() as batch:
+                batch.insert(
+                    table=self.table_name,
+                    columns=self._table_fields,
+                    values=values_batch,
+                )
+
+    def delete(self, documents: List[Document]):
+        """Delete documents from the table."""
+        database = self.client.instance(self.instance).database(self.database)
+        keys = [[doc.page_content] for doc in documents]
+        for keys_batch in _batch(keys, MUTATION_BATCH_SIZE):
+            docs_to_delete = KeySet(keys=keys_batch)
```

Comment:
Sorry updating this logic, We should update this to check the entire doc since they technically can hook it up to any table and page content might not be unique

Reply:
Updated.